packages/flutter/lib/src/material/app.dart

```diff
@@ -287,12 +293,43 @@ class _MaterialScrollBehavior extends ScrollBehavior {
 }
 
 class _MaterialAppState extends State<MaterialApp> {
+  LocalizationsDelegate _localizationsDelegate;
   HeroController _heroController;
 
   @override
   void initState() {
     super.initState();
     _heroController = new HeroController(createRectTween: _createRectTween);
+    _localizationsDelegate = _createLocalizationsDelegate();
+  }
+
+  @override
+  void didUpdateWidget(MaterialApp old) {
+    super.didUpdateWidget(old);
+    final LocalizationsDelegate delegate = widget.localizationsDelegate;
+    final LocalizationsDelegate oldDelegate = old.localizationsDelegate;
+    if (delegate == null && oldDelegate != null
+        || delegate != null && delegate.shouldReload(oldDelegate))
```

Comment:
probably best to put parens around the two sides of the `||` here, i'm not confident of how this will actually fall out. (Also, add a test to verify that it falls out correctly.)